templates/quickstart-hashicorp-vault.template

```diff
@@ -312,364 +578,397 @@ Resources:
           files:
             "/etc/vault.d/vault.hcl":
               source:
-                Fn::Sub:
-                - https://s3.amazonaws.com/${QSS3BucketName}/${QSS3KeyPrefix}scripts/vault.hcl
-                - QSS3BucketName:
-                    Ref: QSS3BucketName
-                  QSS3KeyPrefix:
-                    Ref: QSS3KeyPrefix
+                !Sub
+                  - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}scripts/vault.hcl'
+                  - S3Region: !If [UsingDefaultBucket, !Ref 'AWS::Region', !Ref QSS3BucketRegion]
+                    S3Bucket: !If [UsingDefaultBucket, !Sub '${QSS3BucketName}-${AWS::Region}', !Ref QSS3BucketName]
               mode: '000700'
               owner: root
               group: root
             "/etc/systemd/system/vault.service":
               source:
-                Fn::Sub:
-                - https://s3.amazonaws.com/${QSS3BucketName}/${QSS3KeyPrefix}scripts/vault.service
-                - QSS3BucketName:
-                    Ref: QSS3BucketName
-                  QSS3KeyPrefix:
-                    Ref: QSS3KeyPrefix
+                !Sub
+                  - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}scripts/vault.service'
+                  - S3Region: !If [UsingDefaultBucket, !Ref 'AWS::Region', !Ref QSS3BucketRegion]
+                    S3Bucket: !If [UsingDefaultBucket, !Sub '${QSS3BucketName}-${AWS::Region}', !Ref QSS3BucketName]
               mode: '000700'
               owner: root
               group: root
           commands:
             01_chmod_and_chown_vault:
-              command: chmod 0755 /usr/bin/vault && chown root:root /usr/bin/vault
+              command: chmod 0755 /usr/bin/vault && chown vault:vault /usr/bin/vault
         run_vault:
           commands:
             01_register_vault_binary:
-              command: service vault start
+              command: systemctl daemon-reload && systemctl enable vault && systemctl start vault
```

Comment:
This should be 3 separate commands in order:
01_reload_daemon:
  (...)
02_enable_vault_service:
  (...)
03_start_vault_service:
  (..)
```

Reply:
Currently not in use. Will adjust this for future use.